### PR TITLE
No promise network - bug in flows (DEBUG)

### PR DESCRIPTION
**Pull request by qballer**
Some operations of flows stream in bulk when they are handled by the p-queue. The following PR should fix that and allow to stream updates in the UI, rather than get lots of messages one after the other. This should make the flow execution progress make sense in the UI.
  
- [X]  Remove flows from network
- [x] support `postFlow`  from promise.
- [x] use `mergeMap`/`zip`/`map`/`what-not` operators  to replace `p-queue`
- [x] new ui layer 
flows

---
This change is [Reviewable](https://reviewable.io/reviews/teambit/bit/2602)

### CODE REVIEW

File: src/extensions/flows/flow/flow.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class Flow {
     const endTime = new Date();
     subject[isError ? 'error' : 'next']({
       type: 'flow:result',
-      id: capsule.component.id,
+      id: capsule.component.id.toString(),
```
Comment (davidfirst) [inline src/extensions/flows/flow/flow.ts:72]: @qballer , I changed it before to be the BitId, not a string by purpose. 
We MUST NOT use strings. Here, in particular I need it to be BitId when I use it later.

Comment (qballer) [reply src/extensions/flows/flow/flow.ts:72]: I see.

File: e2e/harmony/flows.e2e.4.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { IS_WINDOWS } from '../../src/constants';
 
 chai.use(require('chai-fs'));
 
-describe('flows functionality', function() {
+describe.only('flows functionality', function() {
```
Comment (davidfirst) [inline e2e/harmony/flows.e2e.4.ts]: @qballer , don't forget the only :)

File: src/extensions/isolator/isolator.ts [hunk 3]
```diff
@@ -86,13 +86,15 @@ export default class Isolator {
...
-        after[i].packageManager === '' ||
-        !isOldPackageManager(after[i].packageManager, config, packageManager)
-      );
-    });
+    const toInstall = capsules
+      .filter((item, i) => {
+        return (
+          !equals(before[i], after[i]) ||
+          after[i].packageManager === '' ||
+          !isOldPackageManager(after[i].packageManager, config, packageManager)
```
Comment (davidfirst) [inline src/extensions/isolator/isolator.ts:94]: @qballer , let's change this function name on your branch. Maybe isPackageManagerIsTheSame

**Pull request merged**